airflow/operators/python.py

```diff
@@ -412,140 +427,113 @@ def __init__(  # pylint: disable=too-many-arguments
+        # Remove unserializable objects
+        # otherwise "KeyError: 'Variable __getstate__ does not exist'" would be raised.
+        self.op_kwargs.pop('var', None)
+        # otherwise "TypeError: cannot serialize '_io.FileIO' object" would be raised.
+        self.op_kwargs.pop('task_instance', None)
+        self.op_kwargs.pop('ti', None)
+
+        if self.system_site_packages or 'apache-airflow' in self.requirements:
+            # All can be serialized expecting it to run in an airflow env.
+            return
+
+        # Not access to host packages and no apache-airflow installed.
+        # Remove airflow specific context
+        # otherwise "ModuleNotFoundError: No module named 'airflow'" would be raised.
+        self.op_kwargs.pop('macros', None)
+        self.op_kwargs.pop('conf', None)
+        self.op_kwargs.pop('dag', None)
+        self.op_kwargs.pop('dag_run', None)
+        self.op_kwargs.pop('task', None)
+
+        if 'pendulum' in self.requirements and 'lazy_object_proxy' in self.requirements:
+            # ..but pendulum is installed so keep pendulum date objects
+            # Note: 'lazy_object_proxy' is needed to work.
+            return
+
+        # No pendulum is installed either. So remove pendulum specific context.
+        # otherwise "ModuleNotFoundError: No module named 'pendulum'" would be raised.
+        self.op_kwargs.pop('execution_date', None)
+        self.op_kwargs.pop('next_execution_date', None)
+        self.op_kwargs.pop('prev_execution_date', None)
+        self.op_kwargs.pop('prev_execution_date_success', None)
+        self.op_kwargs.pop('prev_start_date_success', None)
 
     def _write_string_args(self, filename):
-        # writes string_args to a file, which are read line by line
         with open(filename, 'w') as file:
             file.write('\n'.join(map(str, self.string_args)))
 
-    def _write_args(self, input_filename):
-        # serialize args to file
-        if self._pass_op_args():
-            with open(input_filename, 'wb') as file:
-                arg_dict = ({'args': self.op_args, 'kwargs': self.op_kwargs})
-                if self.use_dill:
-                    dill.dump(arg_dict, file)
-                else:
-                    pickle.dump(arg_dict, file)
-
-    def _read_result(self, output_filename):
-        if os.stat(output_filename).st_size == 0:
+    def _write_script(self, filename):
+        with open(filename, 'w') as file:
+            python_code = render_virtualenv_script(
+                jinja_context=dict(
+                    op_args=self.op_args,
+                    op_kwargs=self.op_kwargs,
+                    pickling_library=self.pickling_library.__name__,
+                    python_callable=self.python_callable.__name__,
+                    python_callable_source=dedent(inspect.getsource(self.python_callable))
+                )
+            )
+            self.log.debug('Writing code to file\n %s', python_code)
```

Comment:
```suggestion
            self.log.debug('Writing code to file: %s', python_code)
```
WDYT?

Reply:
I added `:` but the new line should be there I think.